### DISCUSSION TITLE
Handling more complex CQRS transactions

### DISCUSSION THREAD

**Discussion by gabs247** [jasontaylordev/CleanArchitecture#543] [Q&A]
Thank you for providing this template.

I can see how mediatR + CQRS makes the CA more easily testable, and the simple examples demonstrates how it works.

But I'm trying to think ahead how it might work with more complex commands/queries? It could be the case where specific logic is repeated in multiple request handlers, say if Entity A encapsulates Entity B or if there are multiple commands in a transaction?

I'm wondering, what the best way to tackle that would be? Or am I misunderstanding something?

**Comment by mweel1** [comment]
Commands should not call other commands.    Use a service layer if there is code that multiple commands need to share.

**Reply by gabs247** [reply]
Thanks, that makes sense.
